pybamm/processed_variable.py

```diff
@@ -0,0 +1,69 @@
+#
+# Processed Variable class
+#
+import numpy as np
+import scipy.interpolate as interp
+
+
+class ProcessedVariable(object):
+    """
+    An object that can be evaluated at arbitrary (scalars or vectors) t and x, and
+    returns the (interpolated) value of the base variable at that t and x.
+
+    Parameters
+    ----------
+    base_variable : :class:`pybamm.Symbol`
+        A base variable with a method `evaluate(t,y)` that returns the value of that
+        variable. Note that this can be any kind of node in the expression tree, not
+        just a :class:`pybamm.Variable`.
+        When evaluated, returns an array of size (m,n)
+    t_sol : array_like, size (m,)
+        The time vector returned by the solver
+    y_sol : array_like, size (m, k)
+        The solution vector returned by the solver. Can include solution values that
+        other than those that get read by base_variable.evaluate() (i.e. k>=n)
+    mesh : :class:`pybamm.Mesh`
+        The mesh used to solve, used here to calculate the reference x values for
+        interpolation
+    interp_kind : string
+        The method to use for interpolation
+
+    """
+
+    def __init__(self, base_variable, t_sol, y_sol, mesh=None, interp_kind="linear"):
+        if mesh is not None and base_variable.domain != []:
+            # Process the discretisation to get x values
+            x_sol = np.concatenate([mesh[dom][0].nodes for dom in base_variable.domain])
+            len_x = len(x_sol)
+        else:
+            # No discretisation provided, or variable has no domain (function of t only)
```

Comment:
a user could forget to pass in a mesh for a variable discretised on a domain. Should raise an exception in this case.